docker/models/containers.py

```diff
@@ -0,0 +1,144 @@
+        )
+
+    def export(self, *args, **kwargs):
+        return self.client.api.export(self.id, *args, **kwargs)
+
+    def get_archive(self, *args, **kwargs):
+        return self.client.api.get_archive(self.id, *args, **kwargs)
+
+    def kill(self, *args, **kwargs):
+        return self.client.api.kill(self.id, *args, **kwargs)
+
+    def logs(self, *args, **kwargs):
+        return self.client.api.logs(self.id, *args, **kwargs)
+
+    def pause(self, *args, **kwargs):
+        return self.client.api.pause(self.id, *args, **kwargs)
+
+    def put_archive(self, *args, **kwargs):
+        return self.client.api.put_archive(self.id, *args, **kwargs)
+
+    def remove(self, *args, **kwargs):
+        return self.client.api.remove_container(self.id, *args, **kwargs)
+
+    def rename(self, *args, **kwargs):
+        return self.client.api.rename(self.id, *args, **kwargs)
+
+    def resize(self, *args, **kwargs):
+        return self.client.api.resize(self.id, *args, **kwargs)
+
+    def restart(self, *args, **kwargs):
+        return self.client.api.restart(self.id, *args, **kwargs)
+
+    def start(self, *args, **kwargs):
+        return self.client.api.start(self.id, *args, **kwargs)
+
+    def stats(self, *args, **kwargs):
+        return self.client.api.stats(self.id, *args, **kwargs)
+
+    def stop(self, *args, **kwargs):
+        return self.client.api.stop(self.id, *args, **kwargs)
+
+    def top(self, *args, **kwargs):
+        return self.client.api.top(self.id, *args, **kwargs)
+
+    def unpause(self, *args, **kwargs):
+        return self.client.api.unpause(self.id, *args, **kwargs)
+
+    def update(self, *args, **kwargs):
+        return self.client.api.update_container(self.id, *args, **kwargs)
+
+    def wait(self, *args, **kwargs):
+        return self.client.api.wait(self.id, *args, **kwargs)
+
+
+class ContainerCollection(Collection):
+    model = Container
+
+    def run(self, image, command=None, stdout=True, stderr=False, remove=False,
+            **kwargs):
+        if isinstance(image, Image):
+            image = image.id
+        detach = kwargs.get("detach", False)
+        if detach and remove:
+            raise RuntimeError("The options 'detach' and 'remove' cannot be "
+                               "used together.")
+        try:
+            container = self.create(image, command, **kwargs)
+        except ImageNotFound:
+            self.client.images.pull(image)
+            container = self.create(image, command, **kwargs)
+
+        container.start()
+
+        if detach:
+            return container
+
+        exit_status = container.wait()
+        if exit_status != 0:
+            stdout = False
+            stderr = True
+        out = container.logs(stdout=stdout, stderr=stderr)
+        if remove:
+            container.remove()
+        if exit_status != 0:
+            raise ContainerError(container, exit_status, command, image, out)
```

Comment:
Do we want to treat this as an exceptional occurrence? I don't have my mind made up on it yet, but I'd like to discuss the pros and cons of this versus returning an `(exit_status, logs)` tuple and letting the consumer handle the situation on their own terms.